automation/src/test/scala/org/broadinstitute/dsde/test/api/WorkspacesAzureApiSpec.scala

```diff
@@ -154,6 +155,62 @@ class AzureWorkspacesSpec extends AnyFlatSpec with Matchers with CleanUp {
     }
   }
 
+  it should "allow sharing a workspace" in {
+    implicit val token = owner.makeAuthToken()
+    withTemporaryAzureBillingProject(azureManagedAppCoordinates) { projectName =>
+      val workspaceName = generateWorkspaceName()
+      Rawls.workspaces.create(
+        projectName,
+        workspaceName,
+        Set.empty,
+        Map("disableAutomaticAppCreation" -> "true")
+      )
+      try {
+        val response = workspaceResponse(Rawls.workspaces.getWorkspaceDetails(projectName, workspaceName))
+        response.workspace.name should be(workspaceName)
+        response.workspace.cloudPlatform should be(Some(WorkspaceCloudPlatform.Azure))
+
+        // nonOwner is not a member of the workspace, should not be able to write
+        val userToken = nonOwner.makeAuthToken()
+        eventually {
+          intercept[RestException] {
+            Orchestration.workspaces.setAttributes(projectName, workspaceName, Map("foo" -> "bar"))(userToken)
+          }
+        }
+
+        // Make nonOwner a writer
+        Orchestration.workspaces.updateAcl(
+          projectName,
+          workspaceName,
+          nonOwner.email,
+          WorkspaceAccessLevel.Writer,
+          Some(false),
+          Some(false)
+        )
+        // Verify can write to workspace
+        Orchestration.workspaces.setAttributes(projectName, workspaceName, Map("foo" -> "bar"))(userToken)
+
+        // Remove write access
+        Orchestration.workspaces.updateAcl(
+          projectName,
+          workspaceName,
+          nonOwner.email,
+          WorkspaceAccessLevel.NoAccess,
+          Some(false),
+          Some(false)
+        )
+        eventually {
+          intercept[RestException] {
+            Orchestration.workspaces.setAttributes(projectName, workspaceName, Map("key" -> "value"))(userToken)
```

Comment:
It might make sense to verify you cannot get a SAS.

Reply:
Great idea-- let me try switching to that.